lib/active_model/serializer/fieldset.rb

```diff
@@ -18,14 +18,23 @@ def fields_for(type)
       attr_reader :raw_fields
 
       private
-
       def parsed_fields
+        # TODO: this is not very flexible.
+        # - we should probably support symbol, array, hash, etc
+        #
+        # JSONAPI::IncludeDirective supports these, but also changes the resulting structure.
+        # is it worth it to duplicate some of that functionality?
+        # or should IncludeDirective be modified to only allow one layer of options?
```

Comment:
example:

```ruby
# the root type *should* be assumed (it's currently not with JSON API
fields: :author
fields :author, :comments
fields :access_token # from the docs
```
docs: https://github.com/rails-api/active_model_serializers/blob/master/docs/general/rendering.md#fields
and: https://github.com/rails-api/active_model_serializers/blob/master/docs/general/fields.md 

There is an inconsistency between APIs in the different adapters.

JSON API requires the root key in the fields specification, whereas the other adapters do not require the root. 

So, maybe this PR should focus on unifying this between all adapters.
 - @bf4, @beauby